### PR TITLE
Remove duplication of apps

### PR DESCRIPTION
**Pull request by yulianedyalkova**
Do not attempt to create the same app twice if it is referenced more than once in the releases

### CODE REVIEW

File: service/controller/key/key.go [hunk 1]
```diff
@@ -43,18 +43,33 @@ func ExtractOperators(comps []releasev1alpha1.ReleaseSpecComponent) []releasev1a
...
+		for _, component := range release.Spec.Components {
+			if IsOperator(component) && (operators[BuildAppName(component)] == releasev1alpha1.ReleaseSpecComponent{}) {
+				operators[BuildAppName(component)] = component
+			}
+		}
+	}
+	return operators
+}
+
+func IsOperator(component releasev1alpha1.ReleaseSpecComponent) bool {
```
Comment (MarcelMue) [inline service/controller/key/key.go:59]: We should use this function in line 39 in the `ExtractOperators` function.

Comment (yulianedyalkova) [reply service/controller/key/key.go:59]: That's why I added the function, I just forgot to do it.

File: service/controller/key/key.go [hunk 2]
```diff
@@ -36,25 +36,40 @@ func BuildAppName(operator releasev1alpha1.ReleaseSpecComponent) string {
...
 	}
 	return comp.Version
 }
 
-func AppReferenced(operators []releasev1alpha1.ReleaseSpecComponent, app applicationv1alpha1.App) bool {
-	for _, operator := range operators {
-		if BuildAppName(operator) == app.Name && GetOperatorRef(operator) == app.Spec.Version {
-			return true
-		}
+func AppReferenced(operators map[string]releasev1alpha1.ReleaseSpecComponent, app applicationv1alpha1.App) bool {
```
Comment (MarcelMue) [inline service/controller/key/key.go:70]: In a follow up - let's sort functions in this file alphabetically.

Comment (yulianedyalkova) [reply service/controller/key/key.go:70]: That was a part of my next PR already

**Pull request merged**